src/main.rs

```diff
@@ -137,6 +139,23 @@ fn main() -> Result<i32> {
         }?);
     }
 
+    let mut session = match (&args.session, &args.session_read_only, &url.host_str()) {
+        (Some(name_or_path), None, Some(host)) | (None, Some(name_or_path), Some(host)) => {
+            let read_only = args.session_read_only.is_some();
+            Some(
+                Session::load_session(host, name_or_path, read_only)
+                    .context(format!("couldn't load session {}", name_or_path))?,
+            )
+        }
+        (_, _, _) => None,
+    };
+
+    if let Some(ref s) = session {
+        for (key, value) in s.headers()?.iter() {
+            headers.entry(key).or_insert(value.clone());
```

Comment:
Nitpick: if the key already exists, this clones unnecessarily, so `.or_insert_with()` is more efficient.